packages/framework/presence/src/notificationsManager.ts

```diff
@@ -170,8 +170,16 @@ class NotificationsManagerImpl<
 			Key,
 			InternalTypes.ValueRequiredState<InternalTypes.NotificationType>
 		>,
-		_initialSubscriptions: Partial<NotificationSubscriptions<T>>,
-	) {}
+		initialSubscriptions: Partial<NotificationSubscriptions<T>>,
+	) {
+		// Add event listeners provided at instantiation
+		for (const [nameString, value] of Object.entries(initialSubscriptions)) {
+			// Without schema validation, we don't know that the args are the correct type.
+			// For now we assume the user is sending the correct types and there is no corruption along the way.
```

Comment:
I don't think this comment is correct.

Reply:
The casts need necessary, but are there better types to use?

Reply:
Use this for the loop:
```typescript
		for (const subscriptionName of recordKeys(initialSubscriptions)) {
			// Lingering Event typing issues with Notifications specialization requires
			// this cast. The ony thing that really matters is that name is a string.
			const name = subscriptionName as keyof Events<NotificationSubscriptions<T>>;
			const value = initialSubscriptions[subscriptionName];
			// This check should not be needed while using exactOptionalPropertyTypes, but
			// typescript appears to ignore that with Partial<>. Good to be defensive
			// against callers sending `undefined` anyway.
			if (value !== undefined) {
				this.notificationsInternal.on(name, value);
			}
```
`Object.entries` loses value typing; so, enumerating keys and getting the value will preserve that aspect.
`Object.key` loses the key typing as well, but we can adjust for that with:
```typescript
/**
 * Object.keys retyped to support specific records keys and
 * branded string-based keys.
 */
const recordKeys = Object.keys as <K extends string>(o: Partial<Record<K, unknown>>) => K[];
```
(add locally above the class)
That isn't quite enough - hence cast and comment that maybe can go away some day.